pkg/lib/database/filter.go

```diff
@@ -39,6 +52,64 @@ func MakeFiltersAnd(protoFilters []*model.BlockContentDataviewFilter, store Obje
 	return and, nil
 }
 
+func makeFilters(protoFilters []*model.BlockContentDataviewFilter,
+	store ObjectStore,
+	operator model.BlockContentDataviewFilterOperator,
+	spaceID string,
+) (Filter, error) {
+	switch operator {
+	case model.BlockContentDataviewFilter_No:
+		return MakeFilter(spaceID, protoFilters[0], store)
+	case model.BlockContentDataviewFilter_Or:
+		return makeFilterOr(protoFilters, store, spaceID)
+	case model.BlockContentDataviewFilter_And:
+		return makeFilterAnd(protoFilters, store, spaceID)
+	}
+	return nil, nil
+}
+
+func makeFilterAnd(filters []*model.BlockContentDataviewFilter, store ObjectStore, spaceId string) (Filter, error) {
+	var and FiltersAnd
+	for _, filter := range filters {
+		if len(filter.NestedFilters) == 0 {
+			f, err := MakeFilter(spaceId, filter, store)
+			if err != nil {
+				return nil, err
+			}
+			and = append(and, f)
+			continue
+		}
+		ns := TransformQuickOption(filter.NestedFilters, nil)
```

Comment:
Maybe change signature of `TransfortmQuickOption` to 
```
TransformQuickOption(protoFilter *model.BlockContentDataviewFilter, loc *time.Location) *model.BlockContentDataviewFilter
```
and move it to `MakeFilter`?

It will guarantee that every filter will have properly handled "quick option" and will reduce complexity

Reply:
Fixed